sdk/monitor/Azure.Monitor.Ingestion/tests/IngestionClientTest.cs

```diff
@@ -0,0 +1,227 @@
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Azure.Monitor.Ingestion.Tests
+{
+    public class IngestionClientTest : RecordedTestBase<IngestionClientTestEnvironment>
+    {
+        public IngestionClientTest(bool isAsync) : base(isAsync)
+        {
+        }
+
+        /* please refer to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/TemplateClientLiveTests.cs to write tests. */
+
+        [RecordedTest]
+        public void TestOperation()
+        {
+            Assert.IsTrue(true);
+        }
+
+        #region Helpers
+
+        private static BinaryData GetContentFromResponse(Response r)
+        {
+            // Workaround azure/azure-sdk-for-net#21048, which prevents .Content from working when dealing with responses
+            // from the playback system.
+
+            MemoryStream ms = new MemoryStream();
+            r.ContentStream.CopyTo(ms);
+            return new BinaryData(ms.ToArray());
+        }
+        #endregion
+
+        [Test]
+        public async Task ValidInputFromObjectAsJson()
+        {
+            ClientSecretCredential clientSecretCrendential = new ClientSecretCredential("<tenantId>", "<cliendId>",  "<clientSecret>");
+            var dcrImmutableId = "<dcrImmutableId>";
+            var dcrEndpoint = "https://nibhati-dce-ku6s.westus2-1.ingest.monitor.azure.com";
+            var currentTime = DateTimeOffset.UtcNow.ToString("O");
+
+            BinaryData data = BinaryData.FromObjectAsJson(
+                // Use an anonymous type to create the payload
+                new[] {
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer1",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user1",
+                            TimeZone = "Pacific Time",
+                            Level = 4,
+                            CounterName = "AppMetric1",
+                            CounterValue = 15.3
+                        }
+                    },
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer2",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user2",
+                            TimeZone = "Central Time",
+                            Level = 3,
+                            CounterName = "AppMetric1",
+                            CounterValue = 23.5
+                        }
+                    },
+                });
+
+            LogsIngestionClient client = new(
+                new Uri(dcrEndpoint),
+                clientSecretCrendential);
+
+            // Make the request
+            Response response = client.Upload(dcrImmutableId, "Custom-MyTableRawData", RequestContent.Create(data)); //takes StreamName not tablename
```

Comment:
You should pull the table name/stream name into the environment settings too if they'll vary based on how you set things up

Reply:
It's better to use the async methods in our samples and the test framework will automagically rewrite them int o sync code per https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#sync-async-tests